src/psyclone/psyir/transformations/increase_rank_loop_arrays_trans.py

```diff
@@ -99,28 +98,30 @@ class IncreaseRankLoopArraysTrans(Transformation):
     '''
     def __str__(self):
         return ("Increases the Rank of the supplied arrays by the iteration "
-                "space of the given loop, and update all its references")
-
-    def validate(self, node, options=None):
+                "space of the given loop, and update all references to those"
+                " arrays.")
+
+    def validate(
+        self,
+        node: Loop,
+        arrays: Optional[list[Union[Symbol, str]]] = None
+    ):
         ''' Checks that the supplied node is a valid target.
 
-        :param node: target PSyIR node.
-        :type node: :py:class:`psyclone.psyir.nodes.Loop`
-        :param options: a dictionary with options for transformations.
-        :type options: Optional[Dict[str, Any]]
+        :param node: target Loop node.
+        :param arrays: list of arrays that will have the rank increased.
 
         :raises TransformationError: if the node is not a Loop.
         :raises TransformationError: if the node is not inside a Routine.
-        :raises TransformationError: the array option has not been provided.
-        :raises TransformationError: the given array name  or the
+        :raises TransformationError: if the target node does not have static
+            loop bounds (which the dimension declaration needs).
+        :raises TransformationError: if no array names are provided.
+        :raises TransformationError: the given array name or the
             symbol is not local or not an array.
-        :raises TransformationError: the array is references inside a
-            CodeBlock.
+        :raises TransformationError: if any of the arrays are referenced inside
+            a CodeBlock.
 
         '''
-        if options is None:
-            options = {}
-
         if not isinstance(node, Loop):
```

Comment:
I'm still getting to grips with this myself but I *think* validate should just have the **kwargs argument and then call `get_option` and `validate_options` - see https://psyclone.readthedocs.io/en/latest/developer_guide/transformations.html#moving-to-the-new-transformations-options

Reply:
I'm happy to keep the specific checks that you already have but I think we still then need a following call to `validate_options` since that checks for any unsupported options supplied by the caller. Is that correct @LonelyCat124 ? Also, we need to change the signature of `validate` so that it just has **kwargs. So, I think we'd call `get_option` to get the "arrays" option, then do our specific checks with explicit error messages, then call `validate_options`.